### ISSUE TITLE
Feature request: option to put all semver-compatible upgrades in a single big PR so I can merge them all at once

### ISSUE THREAD

**Issue by jedwards1211** [greenkeeperio/greenkeeper#514]
I have a lot of projects using greenkeeper, and since it makes a separate PR for each dependency in each project, I have 100s of PRs from greenkeeper, which isn't really feasible for me to keep up with.

I think the solution is an option to group all of the semver-compatible package upgrades together in one PR (continuously commiting upgrades to that branch as necessary).  If PRs for major package upgrades are still desired I think it would make sense for them to remain in separate, single-package PRs.

**Comment by gr2m** [comment]
I could imagine that such an option would cause other problems. You would end up with one big PR that updates many different dependencies. If the tests break, how would you know which one of the updates broke it?

Are these 100s of PRs on public repositories / organisations? I’d like to have a look why you have that many. It really shouldn't be that much, I maintain a ton of repositories myself and have Greenkeeper enabled on most of them, and it’s maybe a few pull requests per week, as long as you define version ranges using `^` and not fixed version numbers

**Comment by jedwards1211** [comment]
@gr2m if the tests broke, I would look at the CI build and see if I could figure out what the problem was.  I could be wrong, but I assume I would like that process better than a bunch of little PRs.

Unfortunately many of the repos are private...many are based off of my es2015-library-skeleton and react-library-skeleton projects which have a number of build system deps like eslint, babel, various plugins for each, flow, istanbul, nyc, mocha, chai...you get the picture.

Maybe part of the reason I feel overwhelmed by it and often put off going through the PRs though is I've had intermittent failures due to coveralls upload failing, so greenkeeper files a lot of build failure issues for my projects too.  I'm about to try switching to codecov to alleviate that problem.

**Comment by jedwards1211** [comment]
And actually, I realized going through some right now, just merging one greenkeeper PR sometimes causes another greenkeeper PR to have merge conflicts if the dependencies live right next to each other in the `package.json`, which is really unfortunate.

**Comment by jedwards1211** [comment]
@gr2m another thing Greenkeeper could do is commit the semver-compatible upgrades to the PR branch one-by-one so that if the build fails you can see which package upgrade caused it.  Then when it all passes one could do a squash and merge.

**Comment by gr2m** [comment]
> I've had intermittent failures due to coveralls upload failing

Yeah that is a common problem, be it coveralls, or selenium tests, they tend to fail. There is a [travis_retry](https://docs.travis-ci.com/user/common-build-problems/#travis_retry) command that can help some times. The build should not fail for coveralls though, I would put that in the `after_success` group

> eslint, babel

I know that with these ecosystems the challenge of times is peerDependencies, it is something I was thinking about a lot lately myself. There is a certain order that Greenkeeper should do pull requests in these cases that would reduce the noise a lot. I can’t give you an ETA on it though

> another thing Greenkeeper could do is commit the semver-compatible upgrades to the PR branch one-by-one so that if the build fails you can see which package upgrade caused it. Then when it all passes one could do a squash and merge.

Yeah there could be a way to make it work, it’s just not a priority right now for us to work on it. I want to make the whole project more contributor friendly so folks can start fix their own problems :)
